### PR TITLE
Output escaping

### PR DESCRIPTION
**Pull request by mvanduijker**
#163

### CODE REVIEW

**Comment by pablodip** [comment]
@stof What do you think?

File: src/Pagerfanta/View/Template/Template.php [hunk 1]
```diff
@@ -63,4 +63,9 @@ protected function option($name)
 
         return $this->options[$name];
     }
+
+    protected function escape($value)
+    {
+        return htmlspecialchars($value, ENT_QUOTES, 'UTF-8');
```
Comment (stof) [inline src/Pagerfanta/View/Template/Template.php]: Adding the ENT_SUBSITITUTE flag for PHP 5.4+ for a better output on invalid UTF-8: https://github.com/symfony/symfony/blob/6b025418809a76371d44c8afe5d09c46875789cb/src/Symfony/Component/Debug/ExceptionHandler.php#L441

**Comment by stof** [comment]
Other places inserting user-supplied text also need to apply escaping

**Comment by mvanduijker** [comment]
I left developer settings like classes, next previous texts, etc unescaped. Developers have to apply escaping themselves.

File: src/Pagerfanta/View/Template/Template.php [hunk 2]
```diff
@@ -63,4 +63,9 @@ protected function option($name)
 
         return $this->options[$name];
     }
+
+    protected function escape($value)
```
Comment (pablodip) [inline src/Pagerfanta/View/Template/Template.php]: This can be a static function.

File: tests/Pagerfanta/Tests/View/ViewTestCase.php [hunk 3]
```diff
@@ -61,6 +61,13 @@ protected function renderView($options)
         return $this->view->render($this->pagerfanta, $routeGenerator, $options);
     }
 
+    protected function renderViewNeedsEscaping($options)
+    {
+        $routeGenerator = function ($page) { return '|&'.$page.'|'; };
+
+        return $this->view->render($this->pagerfanta, $routeGenerator, $options);
```
Comment (pablodip) [inline tests/Pagerfanta/Tests/View/ViewTestCase.php]: This could be done by composing the `renderView` method instead of duplicating. :)

**Comment by pablodip** [comment]
How about escaping the text in `generateSpan` and `spanLi`?

Could you also please add some note in the documentation, for people to notice this?

**Comment by mvanduijker** [comment]
I tried that, but you get into a bit more trouble. If the text contains html (like active span in https://github.com/whiteoctober/Pagerfanta/blob/master/src/Pagerfanta/View/Template/TwitterBootstrap3Template.php#L23). this will be escaped. Also all the text which can be configured (like: https://github.com/whiteoctober/Pagerfanta/blob/master/src/Pagerfanta/View/Template/TwitterBootstrapTemplate.php#L20) will be escaped. In the latter case we can get away by using utf8 characters, but freedom for developers to insert html there will be gone. First case is a problem.

**Comment by pablodip** [comment]
Right... Any other thoughts here @stof?

**Comment by mvanduijker** [comment]
Any update on this? Want to finish it :smile:

**Comment by mvanduijker** [comment]
@pablodip any thought on this?

**Pull request closed without merging**